test/evergreen.yml

```diff
@@ -3008,6 +3036,105 @@ tasks:
+          HAVE_UNITTEST: -DHAVE_UNITTEST=1
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/cmake_build"
+          shell: bash
+          include_expansions_in_env:
+            - gcp_sdk_ext_private_key
+            - gcp_sdk_ext_private_key_id
+          script: | 
+            set -o errexit
+            # GCP requires a path to a credentials file for authorization. To not expose the private key within the file, 
+            # use a placeholder private key. The placeholders are then replaced in the command line with the evergreen expansion variables 
+            # and stored in a temporary file. The file is deleted after the unit tests are run.
+            file=$(mktemp --suffix ".json")
+            sed 's|gcp_private_key|'"${gcp_sdk_ext_private_key}"'|;s|gcp_private_id|${gcp_sdk_ext_private_key_id}|' ../test/evergreen/gcp_auth.json >> $file            
+            export GOOGLE_APPLICATION_CREDENTIALS="$file"
+            set -o verbose
+
+            # Run GCP extension unit testing.
+            ext/storage_sources/gcp_store/test/run_gcp_unit_tests
+  - name: azure-tiered-test-small
+    tags: ["pull_request", "python"]
+    commands:
+      - func: "get project"
+      - func: "compile wiredtiger"
+        vars:
+          <<: *configure_flags_tiered_storage_azure
+          cmake_binary: /opt/mongodbtoolchain/v4/bin/cmake
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/cmake_build"
+          shell: bash
+          include_expansions_in_env:
+            - azure_sdk_ext_access_key
+          script: |
+            set -o errexit
+            # Set the Azure credentials using config variable.
+            export AZURE_STORAGE_CONNECTION_STRING="${azure_sdk_ext_access_key}"
+            set -o verbose
+
+            ${test_env_vars|} python3 ../test/suite/run.py -j $(nproc) tiered19
+  - name: gcp-tiered-test-small
+    tags: ["pull_request", "python"]
+    commands:
+      - func: "get project"
+      - func: "install gcp dependencies"
+      - func: "compile wiredtiger"
+        vars:
+          <<: *configure_flags_tiered_storage_gcp
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/cmake_build"
+          shell: bash
+          include_expansions_in_env:
+            - gcp_sdk_ext_private_key
+            - gcp_sdk_ext_private_key_id
+          script: |
+            set -o errexit
+            # GCP requires a path to a credentials file for authorization. To not expose the private key within the file, 
+            # use a placeholder private key. The placeholders are then replaced in the command line with the evergreen expansion variables 
+            # and stored in a temporary file.
+            file=$(mktemp --suffix ".json")
+            sed 's|gcp_private_key|'"${gcp_sdk_ext_private_key}"'|;s|gcp_private_id|${gcp_sdk_ext_private_key_id}|' ../test/evergreen/gcp_auth.json >> $file
```

Comment:
This really needs a comment - for the reason we used `|` instead of the default `/`, for why we need to run this at all, and what the data looks like before and after the replacements. I would also prefer it wasn't repeated - it's not a trivial amount of logic to copy-paste.

Reply:
Let me know what you think about the comments now

Reply:
Thanks, it's much better.